files/jcl/ZWESECUR.jcl

```diff
@@ -310,133 +347,143 @@ $$
-INSERT &ZOWEUSER. UID(&ZOWEUID.) HOME(/tmp) OMVSPGM(/bin/sh)
+INSERT &ZOWEUSER. AUTOUID HOME(/tmp) OMVSPGM(/bin/sh)
 F ACF2,REBUILD(USR),CLASS(P),DIVISION(OMVS)
 *
-* userid for XMEM, cross memory server                              *
+* userid for XMEM cross memory server
+* replace AUTOUID with UID(&XMEMUID.) if AUTOUID is not enabled
 *
 SET LID
 INSERT &XMEMUSER. GROUP(&STCGROUP.)
 SET PROFILE(USER) DIV(OMVS)
-INSERT &XMEMUSER. UID(&XMEMUID.) HOME(/tmp) OMVSPGM(/bin/sh)
+INSERT &XMEMUSER. AUTOUID HOME(/tmp) OMVSPGM(/bin/sh)
 F ACF2,REBUILD(USR),CLASS(P),DIVISION(OMVS)
 *
-* userid for XMEM auxilary cross memory server                      *
-*
-* **** Commented since AUXUSER was set to $XMEMUSER.  ****
-*
+** uncomment if &AUXUSER is changed to not match &XMEMUSER
+** userid for XMEM auxilary cross memory server
+** replace AUTOUID with UID(&AUXUID.) if AUTOUID is not enabled
+**
 * SET LID
 * INSERT &AUXUSER. GROUP(&STCGROUP.)
 * SET PROFILE(USER) DIV(OMVS)
-* INSERT &AUXUSER. UID(&AUXUID.) HOME(/tmp) OMVSPGM(/bin/sh)
+* INSERT &AUXUSER. AUTOUID HOME(/tmp) OMVSPGM(/bin/sh)
 * F ACF2,REBUILD(USR),CLASS(P),DIVISION(OMVS)
+**
+*****
 *
-* operator command F ACF2,REBUILD(USR),CLASS(P)                     *
-* operator command F ACF2,OMVS                                      *
-*
-* started task for ZOWE, main server                                *
+* started task for ZOWE main server
 *
 SET CONTROL(GSO)
 INSERT STC.&ZOWESTC. LOGONID(&ZOWEUSER.) GROUP(&STCGROUP.) +
 STCID(&ZOWESTC.)
 F ACF2,REFRESH(STC)
 *
-* started task for XMEM, cross memory server                        *
+* started task for XMEM cross memory server
 *
 SET CONTROL(GSO)
 INSERT STC.&XMEMSTC. LOGONID(&XMEMUSER.) GROUP(&STCGROUP.) +
 STCID(&XMEMSTC.)
 F ACF2,REFRESH(STC)
 *
-* started task for XMEM auxilary cross memory server                *
+* started task for XMEM auxilary cross memory server
 *
 SET CONTROL(GSO)
 INSERT STC.&AUXSTC. LOGONID(&AUXUSER.) GROUP(&STCGROUP.) +
 STCID(&AUXSTC.)
 F ACF2,REFRESH(STC)
 *
-* operator command F ACF2,REFRESH(STC)                              *
+* DEFINE ZOWE SERVER PERMISIONS ...................................
 *
-* DEFINE ZOWE SERVER PERMISIONS ................................... *
+* permit Zowe main server to use XMEM cross memory server
 *
-* permit Zowe main server to use XMEM, cross memory server          *
+SET RESOURCE(FAC)
+RECKEY ZWES ADD(IS SERVICE(READ) ROLE(&STCGROUP.) ALLOW)
+F ACF2,REBUILD(FAC)
+*
+* permit Zowe main server to create a user's security environment
+* ATTENTION: Defining the BPX.DAEMON or BPX.SERVER profile makes
+*            z/OS UNIX switch to z/OS UNIX level security. This is
+*            more secure, but it can impact operation of existing
+*            applications. Test this thoroughly before activating
+*            it on a production system.
 *
 SET RESOURCE(FAC)
-RECKEY ZWES ADD(IS SERVICE(READ) ROLE(&ADMINGRP.) ALLOW)
+RECKEY BPX ADD(SERVER SERVICE(UPDATE) ROLE(&STCGROUP.) ALLOW)
```

Comment:
The same note as in RACF section: &ZOWEUSER should also have UPDATE access to BPX.DAEMON.
PERMIT BPX.DAEMON CLASS(FACILITY) ACCESS(UPDATE) ID(&ZOWEUSER.)

Reply:
same as before, both users need the permit